src/setup_payload/ManualSetupPayloadGenerator.cpp

```diff
@@ -31,17 +31,60 @@
 
 namespace chip {
 
-static uint32_t shortPayloadRepresentation(const SetupPayload & payload)
+static uint32_t chunk1PayloadRepresentation(const SetupPayload & payload)
 {
-    constexpr int discriminatorOffset = kCustomFlowRequiredFieldLengthInBits;
-    constexpr int pinCodeOffset       = discriminatorOffset + kManualSetupDiscriminatorFieldLengthInBits;
-    uint32_t result                   = payload.requiresCustomFlow ? 1 : 0;
+    /* <1 digit> Represents:
+     *     - <bits 1..0> Discriminator <bits 11.10>
+     *     - <bit 2> VID/PID present flag
+     */
 
-    static_assert(kManualSetupDiscriminatorFieldBitMask <= UINT32_MAX >> discriminatorOffset, "Discriminator won't fit");
-    result |= static_cast<uint32_t>((payload.discriminator & kManualSetupDiscriminatorFieldBitMask) << discriminatorOffset);
+    constexpr int discriminatorShift     = (kPayloadDiscriminatorFieldLengthInBits - kManualSetupDiscriminatorFieldLengthInBits +
```

Comment:
nit: since these are constants, we could name these `kDiscriminatorShift` and `kDiscriminatorMask`

Reply:
Removed unused constants